Sources/TripKitUI/views/map annotations/TKUIAnnotationViewBuilder.swift

```diff
@@ -99,21 +63,14 @@ open class TKUIAnnotationViewBuilder: NSObject {
     } else if let vehicle = annotation as? Vehicle {
       return build(for: vehicle)      
 
-    } else if let timePoint = annotation as? TKUISemaphoreDisplayable, timePoint.semaphoreMode != .none {
-      return buildSemaphore(for: timePoint)
-    
-    } else if let segment = annotation as? TKSegment {
-      return buildSemaphore(for: segment)
+    } else if let semaphore = annotation as? TKUISemaphoreDisplayable {
+      return buildSemaphore(for: semaphore)
 
-    } else if let visit = annotation as? StopVisits {
-      return buildCircle(for: visit)
+    } else if let circle = annotation as? TKUICircleDisplayable {
```

Comment:
would like to know the motivation on making this generic instead of StopVisits, ie use cases.

Reply:
On the mode-by-mode card we want to show things other than stop visits as circles, in particular trip interchanges that are for segments before or after the currently selected segment. The currently selected segment displays those as a "sempahore" and the other segments are displayed just as circles, to keep the UI simpler.